src/Files.Shared/Helpers/PathHelpers.cs

```diff
@@ -14,5 +15,45 @@ public static string Combine(string folder, string name)
 
 			return folder.Contains('/') ? Path.Combine(folder, name).Replace('\\', '/') : Path.Combine(folder, name);
 		}
+
+		public static string FormatName(string path)
+		{
+			string fileName;
+			string rootPath = Path.GetPathRoot(path) ?? string.Empty;
+
+			if (rootPath == path && path.StartsWith(@"\\"))
+			{
+				// Network Share path
+				fileName = path.Substring(path.LastIndexOf(@"\", StringComparison.Ordinal) + 1);
+			}
+			else if (rootPath == path)
+			{
+				// Drive path
+				fileName = path;
+			}
+			else
+			{
+				// Standard file name
+				fileName = Path.GetFileName(path);
+			}
+
+			// Check for link file name
+			if (FileExtensionHelpers.IsShortcutOrUrlFile(fileName))
+				fileName = fileName.Remove(fileName.Length - 4);
+
+			return fileName;
+		}
+
+		public static bool IsSpecialFolder(string path)
```

Comment:
What's considered a special folder? It might be worth adding a comment.

Reply:
Done!